elastic/kibana
# [Fleet] Fix package version comparison in the UI

## Summary

The way we compare package version in the UI was causing a bug for the nginx package `3.10.0` where we are not showing the upgrade button if the `3.9` package is installed for example

That PR fix that by using semver function to compare versions.

## How to test 

1. install nginx 3.9.0 
```
curl --request POST \
  --url http://localhost:5601/api/fleet/epm/packages/nginx-0.3.9 \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'kbn-xsrf: test' \
  --data '{"force": true}'
  ```
1. go to the nginx integration details page you should see the upgrade available label an the upgrade button

<img width="1427" alt="Screen Shot 2021-03-03 at 2 38 03 PM" src="https://user-images.githubusercontent.com/1336873/109862775-d6dbc900-7c2e-11eb-9111-7a200d7ac3a6.png">
<img width="587" alt="Screen Shot 2021-03-03 at 2 37 57 PM" src="https://user-images.githubusercontent.com/1336873/109862779-d80cf600-7c2e-11eb-9bf8-8d5a32bfbaff.png">
